tribits/ci_support/CreateIssueTrackerFromCDashQuery.py

```diff
@@ -0,0 +1,271 @@
+import os
+import sys
+import pprint
+
+import CDashQueryAnalyzeReport as CDQAR
+import cdash_build_testing_date as CDBTD
+from Python2and3 import distItems
+
+g_pp = pprint.PrettyPrinter(indent=2)
+
+
+# Class that is used to create issue trackers for nonpassing tests downloaded
+# from a CDash queryTests.php query.
+#
+# To make this general, the user has to create a class object that is passed
+# in as the argument 'issueTrackerFormatter' in this class that supports
+# the following functions.
+#
+# issueTrackerFormatter.acceptIssueTrackerData(issueTrackerData):
+#
+# This function is given data about the nonpasing tests through an object of
+# type 'IssueTrackerData' as the argument 'issueTrackerData'.  This data is
+# used to create the issue issue tracker text.
+# issueTrackerFormatter.getIssueTrackerText():
+#
+# This function is expected to return the created text for the issue tracker
+# given the data provided in the above 'acceptIssueTrackerData()' function.
+# The format of this text can be anything such as Markdown, RestructredText,
+# JIRA Issue Markup, etc.  The only assumption about formatting is in the HTML
+# table passed in the arugment 'issueTrackerData.testHistoryHtmlTableText'
+# 'acceptIssueTrackerData()' function.
+#
+# The object 'issueTrackerFormatter' can format the issue tracker text anyway
+# it wants.  The object 'issueTrackerFormatter' is stateless w.r.t. this class
+# CreateIssueTrackerFromCDashQueryDriver
+#
+class CreateIssueTrackerFromCDashQueryDriver:
+
+
+  def __init__(self, issueTrackerFormatter, cdashProjectStartTimeUtc=None,
+      usageHelp="", issueTrackerUrlTemplate="", issueTrackerTemplate="",
+    ):
+    self.issueTrackerFormatter = issueTrackerFormatter
+    self.cdashProjectStartTimeUtc = cdashProjectStartTimeUtc
+    self.usageHelp = usageHelp
+    self.issueTrackerUrlTemplate = issueTrackerUrlTemplate
+    self.issueTrackerTemplate = issueTrackerTemplate
+    self.options = None
+
+
+  def runDriver(self):
+
+    print("\n***")
+    print("*** Getting data to create a new issue tracker")
+    print("***\n")
+
+    self.getCmndLineOptions()
+
+    print("Downloading full list of nonpassing tests from CDash URL:\n")
+    print("   "+self.options.nonpassingTestsUrl+"\n")
+
+    nonpassingTestsLOD = self.downloadNonpassingTestsData()
```

Comment:
What's `LOD` short for?

Reply:
LOD == "List of Dicts".   It is a very common acronym in this CDash-related code.  Likely need to define some common acronyms somewhere.